src/test/java/io/vertx/ext/apex/UtilsTest.java

```diff
@@ -50,22 +50,35 @@ public void testPathWithSpaces2() throws Exception {
 
   @Test
   public void testDodgyPath1() throws Exception {
-    assertEquals("/foo///blah", Utils.normalisePath("/foo/../../blah"));
+    assertEquals("/foo/blah", Utils.normalisePath("/foo/../../blah"));
   }
 
   @Test
   public void testDodgyPath2() throws Exception {
-    assertEquals("/foo////blah", Utils.normalisePath("/foo/../../../blah"));
+    assertEquals("/foo/blah", Utils.normalisePath("/foo/../../../blah"));
   }
 
   @Test
   public void testDodgyPath3() throws Exception {
-    assertEquals("/foo//blah", Utils.normalisePath("/foo/../blah"));
+    assertEquals("/foo/blah", Utils.normalisePath("/foo/../blah"));
   }
 
   @Test
   public void testDodgyPath4() throws Exception {
-    assertEquals("//blah", Utils.normalisePath("/../blah"));
+    assertEquals("/blah", Utils.normalisePath("/../blah"));
+  }
+
+  @Test
+  public void testDodgyPathEncoded() throws Exception {
+    assertEquals("/blah", Utils.normalisePath("/%2E%2E%2Fblah"));
+  }
+
+  @Test
+  public void testURLDecode() throws Exception {
+    assertEquals(
+        "/ !\"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\]^_`abcdefghijklmnopqrstuvwxyz{|}~",
+        Utils
+            .normalisePath("/%20%21%22%23%24%25%26%27%28%29%2A%2B%2C%2D%2E%2F%30%31%32%33%34%35%36%37%38%39%3A%3B%3C%3D%3E%3F%40%41%42%43%44%45%46%47%48%49%4A%4B%4C%4D%4E%4F%50%51%52%53%54%55%56%57%58%59%5A%5B%5C%5D%5E%5F%60%61%62%63%64%65%66%67%68%69%6A%6B%6C%6D%6E%6F%70%71%72%73%74%75%76%77%78%79%7A%7B%7C%7D%7E"));
   }
```

Comment:
Looks good. I think we add some more unicode examples e.g. some chinese chars:

%E5%A9%B4%E5%84%BF%E6%9C%8D%E9%A5%B0"